.github/workflows/docker-build.yml

```diff
@@ -58,10 +55,24 @@ jobs:
             type=semver,pattern=v{{major}}
             type=sha
             type=raw,value=${{ env.TAG }}, enable=true
+      - name: Set up build cache
```

Comment:
What are these steps for? They don't seem necessary based on the documentation: https://docs.docker.com/build/ci/github-actions/cache/

Reply:
Workaround buildkit/buildx's lack of integrated caching solution:
https://github.com/reproducible-containers/buildkit-cache-dance/tree/v2.1.2/

Reply:
@jonolehagemo OK, after reading a lot of documentation I now believe that we are caching the result of the build stage of the Docker build in addition to the layers?

It would be helpful to have a description of that in the PR.